### ISSUE TITLE
In debug mode, we should throw assertions when incoming positions are NaN or Inf

### ISSUE THREAD

**Issue by mxgrey** [dartsim/dart#597]
Currently we have assertions which (in debug mode) check that the transformations generated by the kinematics and dynamics are valid transformations, and this is useful for identifying when a bug is occurring. But by the time we're computing the transformation, it can be hard or even impossible to tell what the source of the bug was.

I think if we put in assertions to check that incoming positions (and velocities, accelerations, etc) are not NaN or Inf, it will be much easier to identify what the source of the error is.

**Comment by jslee02** [comment]
Precondition check sounds good. I'd like to keep the current postcondition checks along precondition checks.

**Comment by mxgrey** [comment]
Yes, I agree that we should keep all the existing checks in place.
